src/utils/string-trim.js

```diff
@@ -2,14 +2,24 @@
 // WARNING: this file is used by both the client and the server.
 // Do not use any browser or node-specific API!
 // -------------------------------------------------------------
-/* eslint hammerhead/proto-methods: 2 */
 
 // NOTE: Some websites override the String.prototype.trim method. When we use this function
 // in our scripts, we expect it to have the default behavior. Therefore, in order to protect
-// ourselves from spoofing, we must use our own implementation.
+// ourselves from spoofing, we must use our own implementation. Also we can not use the
+// String.prototype.trim method because on the client-side it is one of a top window and an
+// iframe window.
 
-var strTrim = String.prototype.trim;
+const WHITE_SPACE_REG_EX = /\s/;
 
 export default function (str) {
-    return strTrim.call(str);
+    if (typeof str !== 'string')
+        return str;
+
+    str = str.replace(/^\s\s*/, '');
+
+    var i = str.length;
+
+    while (WHITE_SPACE_REG_EX.test(str.charAt(--i)));
+
+    return str.slice(0, i + 1);
```

Comment:
Why not just `str.replace(/(^\s+)|(\s+$)/g, '');`? Why we need iteration and all additional stuff?